src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```
@@ -572,4 +588,33 @@ public Response getApprovedUsers(@Auth DuosUser duosUser, @PathParam("identifier
       return createExceptionResponse(e);
     }
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_OCTET_STREAM)
+  @Path("/{id}/nihInstitutionalCertification")
+  @RolesAllowed({ADMIN, DATASUBMITTER, CHAIRPERSON, MEMBER})
+  public Response getNihInstitutionalCertification(@Auth DuosUser duosUser, @PathParam("id") Integer id) {
+    User requestingUser = duosUser.getUser();
+    Dataset dataset = datasetService.findDatasetById(requestingUser, id);
+    if (dataset == null) {
+      return Response.status(Response.Status.NOT_FOUND).build();
```

Throw instead of return. The response should be wrapped in a try/catch with a createExceptionResponse in the catch. That will translate all known error types into a consistent error response object.

---

That would be fine if the UI processed our specific error types, but it doesn't in this case.  These don't return JSON, they return application/octet-stream.  This is one of those rare cases where I think we do want to return just error codes and not the DUOS formatted error messages.

---

@rushtong - Code updated to catch any exception.

---

That's a great point. There are two benefits to `createExceptionResponse`, one is to standardize the formatted result (which seems like we do NOT want that here) and the second one is to translate various exceptions that should not go to clients and translate them to something more client friendly, for example, per compliance, we should never send the user a sql exception which might be possible if internal code hits a `Throwable`. Let's see if we can come up with an approach that serves this need.

---

> @rushtong - Code updated to catch any exception.

Thank you - didn't see this when writing the above ☝🏽